devlib/module/cgroups.py

```diff
@@ -180,13 +205,19 @@ def tasks(self, cgroup):
         tasks = {}
         for task in entries:
             tid = task.split(',')[0]
+            if filter_tid and not re.search(filter_tid, tid):
+                continue
             try:
                 tname = task.split(',')[1]
+                if filter_tname and not re.search(filter_tname, tname):
+                    continue
             except: continue
             try:
                 tcmdline = task.split(',')[2]
             except:
                 tcmdline = ''
+            if filter_tcmdline and not re.search(filter_tcmdline, tcmdline):
```

Comment:
This should be in the try block above... we don't want to filter the empty string.

Reply:
I am not sure about this one. If someone wants to filter some pattern in the cmdline, it makes sense to me that a task that has no cmdline should not be in the result.